pkg/services/store/entity/db/dbimpl/dbimpl.go

```diff
@@ -138,6 +140,11 @@ func (db *EntityDB) GetEngine() (*xorm.Engine, error) {
 		return nil, err
 	}
 
+	// register the go_sql_stats_connections_* metrics
+	if err := prometheus.Register(sqlstats.NewStatsCollector("unified_storage", db.engine.DB().DB)); err != nil {
+		db.log.Warn("Failed to register unified storage sql stats collector", "error", err)
+	}
+
```

Comment:
Should this be up in the block where we create the new connection? Or do we need it even when we are piggybacking on the main grafana connection?

Reply:
If we're piggybacking on the grafana connection, then this function should return right away bc of this
```go
if db.engine != nil {
    return db.engine, nil
}
```
at the beginning.

Reply:
nope, check out lines 128-134

Reply:
the check for whether we've already set up db.engine is just a shortcut so you can call GetEngine again and it'll return the engine we configured the first time.

Reply:
ah! gotcha. Fixed it 👍